node/service/src/lib.rs

```diff
@@ -418,28 +503,73 @@ pub fn new_full<RuntimeApi, Executor>(
 		})
 		.collect();
 
-	let (overseer, overseer_handler) = real_overseer(leaves, prometheus_registry.as_ref(), spawner)?;
-	let overseer_handler_clone = overseer_handler.clone();
+	let authority_discovery_service = if role.is_authority() {
+		use sc_network::Event;
+		use futures::StreamExt;
+
+		if authority_discovery_disabled {
+			Err("Authority discovery is mandatory for a validator.")?;
```

Comment:
Rather than failing, we could just override the disable and log that.

Reply:
The `authority_discovery_disabled` flag is going away, making this error handling obsolete. See https://github.com/paritytech/polkadot/pull/1795#discussion_r504007924 for details.